### PR TITLE
Another Pass at Batch Developer Guide

### PR DESCRIPTION
**Pull request by cppwfs**
Hey folks, here is a third pass. Simplified based on what we discussed. Looking for more feedback. 

resolves #3103

### CODE REVIEW

File: content/documentation/4-batch-developer-guides/2-batch/1-simple-task.md [hunk 1]
```diff
@@ -1,41 +1,148 @@
...
+        return args -> {
+            JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+            jdbcTemplate.execute("CREATE TABLE IF NOT EXISTS " +
+                    "BILL_STATEMENTS ( id int, " +
+                    "first_name varchar(50), last_name varchar(50), " +
+                    "minutes int,data_usage int, bill_amount double)");
+        };
+    }
+}
+```
```
Comment (dturanski) [inline content/documentation/4-batch-developer-guides/2-batch/1-simple-task.md]: I think this setup task needs to do more. I'm not sure I would ever write a boot app just to do this. Maybe setup some audit tables, and if the table exists, truncate it, record the event.

File: content/documentation/4-batch-developer-guides/2-batch/1-simple-task.md [hunk 2]
```diff
@@ -1,41 +1,148 @@
...
+   To configure the execution of the billsetuptask application utilize the following arguments:
+
+   1. _spring.datasource.url_ - set the URL to your database instance. In the sample below we are connecting to a mysql `practice` database on our local machine at port 3306.
+   1. _spring.datasource.username_ - the user name to be used for the MySql database. In the sample below it is `root`
+   1. _spring.datasource.password_ - the password to be used for the MySql database. In the sample below it is `password`
+   1. _spring.datasource.driverClassName_ - The driver to use to connect to the MySql database. In the sample below it is `com.mysql.jdbc.Driver'
+
+   ```bash
+   $ java -jar target/billsetuptask-0.0.1-SNAPSHOT.jar \
+   --spring.datasource.url=jdbc:mysql://localhost:3306/practice?useSSL=false \
```
Comment (dturanski) [inline content/documentation/4-batch-developer-guides/2-batch/1-simple-task.md]: can we remove the useSSL flag here?

File: content/documentation/4-batch-developer-guides/2-batch/2-spring-batch.md [hunk 3]
```diff
@@ -8,34 +8,270 @@ description: 'Create a Spring Batch Job'
...
+    1. _spring.datasource.username_ - the user name to be used for the MySql database. In the sample below it is `root`
+    1. _spring.datasource.password_ - the password to be used for the MySql database. In the sample below it is `password`
+    1. _spring.datasource.driverClassName_ - The driver to use to connect to the MySql database. In the sample below it is `com.mysql.jdbc.Driver'
+    1. _spring.datasource.initialization-mode_ - initializes the database with the BILL_STATEMENTS and BILL_USAGE tables required for this app. In the sample below we state that we `always` want to do this. This will not overwrite the tables if they already exist.
+    1. _spring.batch.initialize-schema_ - initializes the database with the tables required for Spring Batch. In the sample below we state that we `always` want to do this. This will not overwrite the tables if they already exist.
+
+    ```bash
+    $ java -jar target/billrun-0.0.1-SNAPSHOT.jar \
+    --spring.datasource.url=jdbc:mysql://localhost:3306/practice?useSSL=false \
```
Comment (dturanski) [inline content/documentation/4-batch-developer-guides/2-batch/2-spring-batch.md]: same question

File: content/documentation/4-batch-developer-guides/2-batch/1-simple-task.md (hunk 1 above)
Comment (cppwfs) [reply content/documentation/4-batch-developer-guides/2-batch/1-simple-task.md]: We can chat with Michael and see if we can dream up something a little more interesting.

File: content/documentation/4-batch-developer-guides/2-batch/1-simple-task.md (hunk 2 above)
Comment (cppwfs) [reply content/documentation/4-batch-developer-guides/2-batch/1-simple-task.md]: It was required when hooking up to the docker instance of mysql.  Else the connection failed.

File: content/documentation/4-batch-developer-guides/2-batch/2-spring-batch.md (hunk 3 above)
Comment (cppwfs) [reply content/documentation/4-batch-developer-guides/2-batch/2-spring-batch.md]: It was required when hooking up to the docker instance of mysql.  Else the connection failed.

**Comment by cppwfs** [comment]
Closing this PR due to merge issues.   Use this PR instead. https://github.com/spring-io/dataflow.spring.io/pull/8

**Pull request closed without merging**